src/vertex_tools.rs

```diff
@@ -0,0 +1,398 @@
+use std::f32;
+
+use nalgebra;
+use nalgebra::base::Vector3;
+use nalgebra::geometry::Point3;
+
+use crate::convert::cast_usize;
+use crate::face_tools;
+use crate::geometry::{Face, Geometry, UnorientedEdge};
+
+/// The Möller–Trumbore ray-triangle intersection algorithm is a fast method for
+/// calculating the intersection of a ray and a triangle in three dimensions
+/// without needing precomputation of the plane equation of the plane containing
+/// the triangle.
+///
+/// https://en.wikipedia.org/wiki/Möller–Trumbore_intersection_algorithm
+/// https://www.scratchapixel.com/lessons/3d-basic-rendering/ray-tracing-rendering-a-triangle/moller-trumbore-ray-triangle-intersection
+#[allow(dead_code)]
+pub fn ray_intersects_triangle(
+    ray_origin: &Point3<f32>,
+    ray_vector: &Vector3<f32>,
+    triangle_vertices: (&Point3<f32>, &Point3<f32>, &Point3<f32>),
+) -> Option<Point3<f32>> {
+    let edge_1_vector = triangle_vertices.1 - triangle_vertices.0;
+    let edge_2_vector = triangle_vertices.2 - triangle_vertices.0;
+    let perpendicular_vector = ray_vector.cross(&edge_2_vector);
+    let determinant = edge_1_vector.dot(&perpendicular_vector);
```

Comment:
I've described all the parts that are straight-forward understandable to me. The rest is some calculation of barycentric coordinates in a tetrahedron and to understand the math I'd have to spend too much time drawing the cases so that i can describe it. 
Also, do you think such descriptions are crucial for our project?